### DISCUSSION TITLE
Are there CASCADE remove relation in pivot table if delete entity ?

### DISCUSSION THREAD

**Discussion by dungkaka** [mikro-orm/mikro-orm#4817] [General]
I have two collections and pivot table: Role, Permission and pivot table RolePermission. Like
```Typescript
@Entity()
class Permission {
    @PrimaryKey({ type: types.smallint })
    id: number;

    @ManyToMany({entity: () => Role, mapped: "permissions"})
    roles = new Collection<Role>(this);
}

@Entity()
class Role {
    @PrimaryKey({ type: types.smallint })
    id: number;

    @ManyToMany({entity: () => Permission, pivotEntity: () => RolePermission})
    permissions = new Collection<Permission>(this);
}

@Entity()
class RolePermission {
    @ManyToOne(() => Role, { primary: true })
    role: Role;

    @ManyToOne(() => Permission, { primary: true })
    permission: Permission;
}

const role = await this.em.findOne(Role, filter, { populate: ["permissions"] });
await this.em.removeAndFlush(role)

```

The target I want is: If I delete a role, it also removes all RolePermission related in pivot table.
I found somewhere information about cascade on ManyToMany, try add Cascade.ALL, but I still throw:
ForeignKeyConstraintViolationException: delete from "role" where "id" in (50) - update or delete on table "role" violates foreign key constraint "role_permission_role_id_foreign" on table "role_permission

NodeJs: 16
MikroORM: 5.8.1

**Comment by B4nan** [comment]
You don't need the `cascade` option, you need `onDelete: 'cascade'` on the FKs, that is how the ORM would generate it if you did not provide the pivot entity yourself, the M:N pivot table would have `on delete cascade` on the FKs, which means that if you remove the target entity, the rows referencing it will be removed automatically by the db engine. `cascade` option is for app-level cascading, and that means that if you remove the `Role` entity, you also remove the `Permission` entity connected to it, it is not about the pivot entity rows at all.

Try this:

```ts
@Entity()
class RolePermission {
    @ManyToOne(() => Role, { primary: true, onDetele: 'cascade' })
    role: Role;

    @ManyToOne(() => Permission, { primary: true, onDetele: 'cascade' })
    permission: Permission;
}
```
